scrapy/core/scheduler.py

```diff
@@ -132,3 +157,33 @@ def _dqdir(self, jobdir):
             if not exists(dqdir):
                 os.makedirs(dqdir)
             return dqdir
+
+
+class Scheduler(BaseScheduler):
+    """
+    Key is the priority of the request (an integer)
+    """
+
+    def _request_key(cls, request):
+        return -request.priority
+
+
+class RoundRobinScheduler(BaseScheduler):
+    """
+    Key is the domain and we round robin, choosing the new request to be from
+    the domain that was requested last.  The default Scheduler sends multiple
+    consecutive requests to a domain if multiple links were discovered and
+    added consecutively.  This scheduler spreads those out that workload.
+
+    Uses `RoundRobinQueue`.  The pqclass parameter and SCHEDULER_PRIORITY_QUEUE
+    are ignored.
+    """
+
+    def __init__(self, dupefilter, jobdir=None, dqclass=None, mqclass=None,
+                 logunser=False, stats=None, pqclass=None):
```

Comment:
I think it should be possible to override `pqclass` argument somehow, now it is silently discarded

Reply:
That's a good point.  I just removed it from the signature of the child class.